### ISSUE TITLE
Apply gradient accumulation fix to DPO/PPO recipes

### ISSUE THREAD

**Issue by SalmanMohammadi** [pytorch/torchtune#2037]

**Comment by andywong-pi** [comment]
Hi @SalmanMohammadi,

I’m curious to know if the gradient accumulation fix has been implemented or if anyone is currently developing the fix. 
If not, could you please describe the issue and outline a potential solution?

I came across a blog post about the gradient accumulation fix for HF’s transformers (https://unsloth.ai/blog/gradient), and I’m wondering if Torchtune might be facing a similar issue.

Thanks!

**Comment by SalmanMohammadi** [comment]
> Hi [@SalmanMohammadi](https://github.com/SalmanMohammadi),
> 
> I’m curious to know if the gradient accumulation fix has been implemented or if anyone is currently developing the fix. If not, could you please describe the issue and outline a potential solution?
> 
> I came across a blog post about the gradient accumulation fix for HF’s transformers (https://unsloth.ai/blog/gradient), and I’m wondering if Torchtune might be facing a similar issue.
> 
> Thanks!

Hey @andywong-pi!

This fix was addressed in our SFT recipes in https://github.com/pytorch/torchtune/pull/1875. This issue was mostly a reminder for me to investigate this issue for our DPO/PPO trainers.
